libcudacxx/test/libcudacxx/std/language.support/support.srcloc/general.pass.cpp

```diff
@@ -0,0 +1,173 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: nvcc-11.1, nvcc-11.2
+// The way we currently compile nvrtc is not working with that test
+// XFAIL: nvrtc
+
+#include <cuda/std/cassert>
+#include <cuda/std/cstdint>
+#include <cuda/std/source_location>
+#include <cuda/std/type_traits>
+
+#include "test_macros.h"
+
+static_assert(cuda::std::is_nothrow_move_constructible<cuda::std::source_location>::value, "support.srcloc.cons (1.1)");
+static_assert(cuda::std::is_nothrow_move_assignable<cuda::std::source_location>::value, "support.srcloc.cons (1.2)");
+#if TEST_STD_VER >= 2014
+static_assert(cuda::std::is_nothrow_swappable<cuda::std::source_location>::value, "support.srcloc.cons (1.3)");
+#endif // TEST_STD_VER >= 2014
+
+ASSERT_NOEXCEPT(cuda::std::source_location());
+#if !defined(TEST_COMPILER_NVCC)
+ASSERT_NOEXCEPT(cuda::std::source_location::current());
+#endif // TEST_COMPILER_NVCC
+
+// Note: the standard doesn't strictly require the particular values asserted
+// here, but does "suggest" them.  Additional tests for details of how the
+// implementation of current() chooses which location to report for more complex
+// scenarios are in the Clang test-suite, and not replicated here.
+
+// A default-constructed value.
+constexpr cuda::std::source_location empty{};
+static_assert(empty.line() == 0, "");
+static_assert(empty.column() == 0, "");
+static_assert(empty.file_name()[0] == '\0', "");
+static_assert(empty.function_name()[0] == '\0', "");
+
+ASSERT_NOEXCEPT(empty.line());
+ASSERT_NOEXCEPT(empty.column());
+ASSERT_NOEXCEPT(empty.file_name());
+ASSERT_NOEXCEPT(empty.function_name());
+static_assert(cuda::std::is_same<unsigned, decltype(empty.line())>::value, "");
+static_assert(cuda::std::is_same<unsigned, decltype(empty.column())>::value, "");
+static_assert(cuda::std::is_same<const char*, decltype(empty.file_name())>::value, "");
+static_assert(cuda::std::is_same<const char*, decltype(empty.function_name())>::value, "");
+
+__device__ _CCCL_CONSTEXPR_GLOBAL cuda::std::source_location device_empty{};
+#if !defined(_CCCL_CUDACC_BELOW_11_3)
```

Comment:
Wouldn't the checks below also hold for cudacc below 11.3?

Reply:
This is specific towards `device_empty` not being a constant expression so we cannot use it within a static_assert